tests/k8st/tests/test_bgp_advert.py

```diff
@@ -0,0 +1,217 @@
+        cluster = self.k8s_client()
+        cluster.create_namespace(client.V1Namespace(metadata=client.V1ObjectMeta(name="bgp-test")))
+        container = client.V1Container(
+            name="nginx",
+            image="nginx:1.7.9",
+            ports=[client.V1ContainerPort(container_port=80)])
+        # Create and configure a spec section
+        template = client.V1PodTemplateSpec(
+            metadata=client.V1ObjectMeta(labels={"app": "nginx"}),
+            spec=client.V1PodSpec(containers=[container]))
+        # Create the specification of deployment
+        spec = client.ExtensionsV1beta1DeploymentSpec(
+            replicas=2,
+            template=template)
+        # Instantiate the deployment object
+        nginx_deployment = client.ExtensionsV1beta1Deployment(
+            api_version="extensions/v1beta1",
+            kind="Deployment",
+            metadata=client.V1ObjectMeta(name="nginx"),
+            spec=spec)
+        api_response = client.ExtensionsV1beta1Api().create_namespaced_deployment(
+            body=nginx_deployment,
+            namespace="bgp-test")
+        _log.debug("Deployment created. status='%s'" % str(api_response.status))
+
+        service = client.V1Service(
+            metadata=client.V1ObjectMeta(
+                name="nginx",
+                labels={"name": "nginx"},
+            ),
+            spec={
+                "ports": [{"port": 80}],
+                "selector": {"app": "nginx"},
+            }
+        )
+        api_response = cluster.create_namespaced_service(
+            body=service,
+            namespace="bgp-test",
+        )
+        _log.debug("Service created. status='%s'" % str(api_response.status))
+
+        # # Modify calico/node to run :master tag and set CALICO_STATIC_ROUTES=10.96.0.0/12
+
+        config.load_kube_config(os.environ.get('KUBECONFIG'))
+        api = client.AppsV1Api(client.ApiClient())
+        node_ds = api.read_namespaced_daemon_set("calico-node", "kube-system", exact=True, export=True)
+        for container in node_ds.spec.template.spec.containers:
+            if container.name == "calico-node":
+                container.image = "calico/node:latest-amd64"
+                route_env_present = False
+                for env in container.env:
+                    if "CALICO_STATIC_ROUTES" in env.name:
+                        route_env_present = True
+                if not route_env_present:
+                    container.env.append({"name": "CALICO_STATIC_ROUTES", "value": "10.96.0.0/12", "value_from": None})
+        api.replace_namespaced_daemon_set("calico-node", "kube-system", node_ds)
+
+        # # Establish BGPPeer from cluster nodes to node-extra using calicoctl
+        subprocess.check_call("""kubectl exec -i -n kube-system calicoctl -- /calicoctl apply -f - << EOF
+apiVersion: projectcalico.org/v3
+kind: BGPPeer
+metadata:
+  name: node-extra.peer
+spec:
+  peerIP: 10.192.0.5
+  asNumber: 64512
+EOF
```

Comment:
Can that `kubectl ... calicoctl ...` be collapsed into a kubectl operation on a CRD?

(Genuinely don't know here; I might be talking rubbish, but I thought we supported something like that.)